From e08fe625655dca6e095b103869e030b79e5d0df2 Mon Sep 17 00:00:00 2001
From: Robert Edmonds <edmonds@users.noreply.github.com>
Date: Sat, 8 Jul 2023 23:14:23 -0400
Subject: [PATCH] protoc-c/c_helpers.h: Remove HAVE_PROTO3 conditional

---
 protoc-c/c_helpers.h | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/protoc-c/c_helpers.h b/protoc-c/c_helpers.h
index 055528bf..062d330b 100644
--- a/protoc-c/c_helpers.h
+++ b/protoc-c/c_helpers.h
@@ -175,14 +175,10 @@ int compare_name_indices_by_name(const void*, const void*);
 // Return the syntax version of the file containing the field.
 // This wrapper is needed to be able to compile against protobuf2.
 inline int FieldSyntax(const FieldDescriptor* field) {
-#ifdef HAVE_PROTO3
-# if GOOGLE_PROTOBUF_VERSION >= 4023000
+#if GOOGLE_PROTOBUF_VERSION >= 4023000
   return FileDescriptorLegacy(field->file()).syntax() == FileDescriptorLegacy::SYNTAX_PROTO3 ? 3 : 2;
-# else
-  return field->file()->syntax() == FileDescriptor::SYNTAX_PROTO3 ? 3 : 2;
-# endif
 #else
-  return 2;
+  return field->file()->syntax() == FileDescriptor::SYNTAX_PROTO3 ? 3 : 2;
 #endif
 }
 
