From c22a08de7cd9fd2395bce27af2390cc0421c285c Mon Sep 17 00:00:00 2001
From: Paul Zander <negril.nx@gmail.com>
Date: Sun, 12 Jan 2025 15:25:52 +0100
Subject: [PATCH] fix -j

parallel_args_file is only defined when -j is passed

Bug: https://bugs.gentoo.org/947920
Signed-off-by: Paul Zander <negril.nx@gmail.com>

diff --git a/copy-firmware.sh b/copy-firmware.sh
index ee0e9c0..4f96e8e 100755
--- a/copy-firmware.sh
+++ b/copy-firmware.sh
@@ -131,9 +131,9 @@ grep -E '^(RawFile|File):' WHENCE | sed -E -e 's/^(RawFile|File): */\1 /;s/"//g'
 done
 if [ "$num_jobs" -gt 1 ]; then
     parallel -j"$num_jobs" -a "$parallel_args_file"
+    echo > "$parallel_args_file"
 fi
 
-echo > "$parallel_args_file"
 # shellcheck disable=SC2162 # file/folder name can include escaped symbols
 grep -E '^Link:' WHENCE | sed -e 's/^Link: *//g;s/-> //g' | while read l t; do
     [ -n "${FIRMWARE_LIST}" ]  && if ! echo "${FIRMWARE_LIST}" | grep -q "${l}"; then continue; fi
-- 
2.48.0

