# https://github.com/rizsotto/Bear/issues/445
diff --git a/test/lit.cfg b/test/lit.cfg
index 6786ce7..cbc99ea 100644
--- a/test/lit.cfg
+++ b/test/lit.cfg
@@ -179,7 +179,7 @@ if which('fakeroot'):
 
 
 # check if valgrind is available
-if which('valgrind'):
+if which('valgrind') and False:
     path = which('valgrind')
     config.substitutions.append(('%{valgrind}', path))
     config.available_features.add('valgrind')
diff --git a/test/cases/compilation/output/compile_cuda.sh b/test/cases/compilation/output/compile_cuda.sh
index 41b8b37..a0ed9bf 100644
--- a/test/cases/compilation/output/compile_cuda.sh
+++ b/test/cases/compilation/output/compile_cuda.sh
@@ -1,7 +1,7 @@
 #!/usr/bin/env sh
 
 # REQUIRES: shell, cuda
-# RUN: cd %T; env CC=%{cuda} %{bear} --verbose --output %t.json -- %{shell} %s
+# RUN: cd %T; env CC=%{cuda} NVCC_APPEND_FLAGS="-allow-unsupported-compiler" %{bear} --output %t.json -- %{shell} %s
 # RUN: assert_compilation %t.json count -eq 2
 # RUN: assert_compilation %t.json contains -file %T/successful_build_1.cu -directory %T
 # RUN: assert_compilation %t.json contains -file %T/successful_build_2.cu -directory %T
 
 
